core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/distinct.kt

```diff
@@ -0,0 +1,39 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import io.kotest.matchers.shouldBe
+import org.jetbrains.kotlinx.dataframe.samples.api.age
+import org.jetbrains.kotlinx.dataframe.samples.api.city
+import org.jetbrains.kotlinx.dataframe.samples.api.firstName
+import org.jetbrains.kotlinx.dataframe.samples.api.name
+import org.junit.Test
+
+class DistinctTests : ColumnsSelectionDslTests() {
+
+    @Test
+    fun distinct() {
+        listOf(
+            df.select { all() },
+            df.select { all().distinct() },
+            df.select { (all() and age and city).distinct() },
+        ).shouldAllBeEqual()
+
+        val cols = dfGroup.getColumnsWithPaths {
+            cols(
+                name.firstName,
+                name.firstName,
+                name.firstName.firstName,
```

Comment:
could we do the same tests but with the paths as Strings?

Reply:
sure!

Thanks, this actually uncovered an issue. I wasn't able to type:
```kotlin
cols(
    "name"["firstName"], // ColumnPath
    "name"["firstName"], // ColumnPath
    "name"["firstName"]["firstName"], // ColumnSet?? huh
).distinct()
```
because I missed a `ColumnPath[String]: ColumnPath` overload. It worked fine like 
`df.select { "name"["firstName"]["firstName"] }` cause `select` doesn't mind whether it gets a `ColumnSet` or a `SingleColumn`, but `vararg` functions like `cols()` do require every argument to be the same type :)